app/views/shared/_new_account.html.erb

```diff
@@ -16,12 +15,14 @@
       <%= standard_field form: f, name: :suffix, type: :text_field, label: "Suffix",
                          options: {value: user.suffix, hide: true} %>
       <%= standard_field form: f, name: :email_address, type: :text_field, label: "Email Address *",
-                         options: {value: user.email_addresses.try(:first).try(:value)} %>
+                         options: {value: email, disabled: email.present?} %>
       <%= standard_field form: f, name: :username, type: :text_field, label: "Username *",
                          options: {value: user.username} %>
 
-      <%= standard_field form: f, name: :password, type: :password_field, label: "Password *" %>
-      <%= standard_field form: f, name: :password_confirmation, type: :password_field, label: "Confirm Password *" %>
+      <% unless signed_in? %>
+        <%= standard_field form: f, name: :password, type: :password_field, label: "Password *" %>
+        <%= standard_field form: f, name: :password_confirmation, type: :password_field, label: "Confirm Password *" %>
+      <% end %>
```

Comment:
While it is true that password sign ups won't be signed in yet, maybe we can be more explicit.  At the top can you say

```ruby
<% is_social_signup = signed_in? %>
```

Then in this block of code say:

```ruby
<% if is_social_signup %>
  <% provider = user.authentications.first.provider %>
  <span class="fa-stack fa-lg">
    <i class="fa fa-square fa-stack-2x <%= provider %>-bkg"></i>
    <i class="fa fa-<%= provider %> fa-stack-1x fa-inverse"></i>
  </span>
  <b>Password managed by <%= provider %></b>
<% else %>
  <%= standard_field form: f, name: :password, type: :password_field, label: "Password *" %>
  <%= standard_field form: f, name: :password_confirmation, type: :password_field, label: "Confirm Password *" %>
<% end >%
```

Might make it more clear to someone reading it later.  The provider stuff I wrote above was not tested, may need fixing.  Going for: 
![image](https://cloud.githubusercontent.com/assets/1001691/13900400/5e4e74e2-edc2-11e5-94dc-5a3a253119da.png)